aten/src/ATen/cpu/vec/vec_base.h

```diff
@@ -538,7 +538,7 @@ struct Vectorized {
     // 1 if the pred is true, otherwise 0.
     Vectorized<T> vector;
     for (int i = 0; i != size(); ++ i) {
-      vector[i] = bool(op(values[i], other.values[i]));
+      vector[i] = static_cast<T>(op(values[i], other.values[i]));
```

Comment:
huh?

Reply:
To pacify MSVC,

```
runner\_work\pytorch\pytorch\build\aten\src\ATen\native\cpu\BinaryOpsKernel.cpp.DEFAULT.cpp
2022-05-14T09:36:24.0841359Z C:/actions-runner/_work/pytorch/pytorch/aten/src/ATen/native/cpu/BinaryOpsKernel.cpp(500): error C2397: conversion from 'bool' to 'scalar_t' requires a narrowing conversion
2022-05-14T09:36:24.0842387Z C:/actions-runner/_work/pytorch/pytorch/aten/src/ATen/native/cpu/BinaryOpsKernel.cpp(491): error C2398: Element '1': conversion from 'bool' to 'const T &' requires a narrowing conversion
2022-05-14T09:36:24.0843023Z         with
2022-05-14T09:36:24.0843317Z         [
2022-05-14T09:36:24.0843578Z             T=float
2022-05-14T09:36:24.0843938Z         ]
2022-05-14T09:36:24.0844601Z C:/actions-runner/_work/pytorch/pytorch/aten/src/ATen/native/cpu/BinaryOpsKernel.cpp(491): error C2398: Element '1': conversion from 'bool' to 'const T &' requires a narrowing conversion
2022-05-14T09:36:24.0845185Z         with
2022-05-14T09:36:24.0845486Z         [
2022-05-14T09:36:24.0845791Z             T=double
2022-05-14T09:36:24.0846061Z         ]
2022-05-14T09:36:24.0846740Z C:/actions-runner/_work/pytorch/pytorch/aten/src/ATen/native/cpu/BinaryOpsKernel.cpp(491): error C2398: Element '1': conversion from 'bool' to 'float' requires a narrowing conversion
2022-05-14T09:36:24.0847493Z Microsoft (R) C/C++ Optimizing Compiler Version 19.28.29337 for x64
```

Ref: https://github.com/pytorch/pytorch/runs/6433697946?check_suite_focus=true#step:6:22091

Reply:
Also note, `vector` holds elements of type T and not bool.

Reply:
I am not really sure. Unfortunately, I don't have a Windows system on me to verify that quickly.